conftest.py

```diff
@@ -314,8 +314,19 @@ def additional_repos(request, hosts):
             host_.ssh(['rm', '-f', repo_file])
 
 @pytest.fixture(scope='session')
-def second_network(request):
-    return request.param
+def second_network(request, host):
+    if request.param is None:
+        pytest.fail("This test requires the --second-network parameter!")
+    network_uuid = request.param
+    pif_uuid = host.xe('pif-list', {'host-uuid': host.uuid, 'network-uuid': network_uuid}, minimal=True)
+    if pif_uuid == "":
+        pytest.fail("The provided --second-network UUID doesn't exist or doesn't have a PIF on master host")
```

Comment:
The network should have an PIF with IP configured on all pools hosts.

Reply:
I know but didn't want to push the tests too far and assumed a normally configured pool would have a PIF and IP for all hosts of the pool if master has.

Reply:
The goal was to avoid common failures at fixture initialisation, but I think it's acceptable if in the case when the pool would be heterogeneous the test would fail later.

Reply:
Fine by me :)